005-github-simplification.Rmd

```diff
@@ -0,0 +1,93 @@
+    -   glue: useful for data science but designed to be zero dep specifically for use by package developers.
+    -   tibble/magrritr: both technically user-facing, but most people use via other packages
+    -   modelr: we don't really want people to use this package.
+    -   googledrive: not really about data analysis or package development.
+    -   vroom is in r-lib, readr is in tidyverse.
+
+    It often takes me multiple attempts to get to the correct URL for these packages.
+
+-   It's painful to move issues across orgs.
+    This affects packages like dplyr where the underlying issue might need to be fixed in vctrs.
+
+-   As we start to use GitHub teams more, managing separate teams across separate orgs will become more painful.
+    It also makes it hard to centralise discussions, and to explain succintly the scope of tidy ups.
+
+## Solutions
+
+There are three possible solutions:
+
+-   Move all r-lib and tidymodels repos to **tidyverse org**.
+-   Create a **new org** and move all r-lib, tidyverse, and tidymodel repos.
+-   Move all r-lib, tidyverse, and tidymodel to the **rstudio org**.
+
+These cases only affect the GitHub; all other websites would stay as is.
+
+### Move to tidyverse org
+
+Pros:
+
+-   Moves smallest number of packages.
+-   Tidyverse has strongest brand identity.
+
+Cons:
+
+-   Makes it more complicated to explain exactly what the tidyverse is. (It's a package; it's a set of packages loaded by `library(tidyverse)`; it's a set of packages installed by `install.packages("tidyverse")`; and it's a GitHub organisation).
+-   No longer a one-to-one relationship between package homepage and GitHub organisation.
+-   Lose ability to bury less-important packages in r-lib; harder to highlight most important packages in tidyverse.
+-   Binds all packages more closely to tidyverse. This might be less appealing for external developers and for other RStudio teams we might hope would also migrate to one org.
+
+### Create new org
+
+Pros:
+
+-   May be easier to persuade other organisations to join if not "tidyverse" branded.
+
+Cons:
+
+-   Will be hard to come up with good name.
+-   In the short term, increases numbers of orgs.
+-   Maximum number of packages need to be moved.
+
+### Move to rstudio org
+
+Pro:
+
+-   Greatest reduction in packages.
+
+-   Easy story to tell (now that RStudio is a PBC, we have 0% concerns about it as long term steward of this code).
+
+Cons:
+
+-   Eliminates veneer of separation between RStudio and tidyverse, and makes it harder to invite non-employee authors and maintainers.
+-   Would fall under RStudio IT practices which are geared towards protecting private data.
+-   Would need very clear reassurance that this isn't RStudio "muscling in" on open source. Regardless of how we communicate this, a few people are likely to generate negative press.
+
+## Open questions
+
+-   Are there any existing workflows that this would break?
```

Comment:
I _think_ that GitHub's automatic forwarding should handle most cases, e.g. cloning, issue linking etc.

That being said I think there are at least a few GitHub APIs that don't automatically forward. I have run into a few cases where usethis functions were not working because I had an Git remote setup with an old URL. Perhaps @jenny knows offhand of a few specifics.

Reply:
As I recall, the friction we encountered in usethis is that one can still access / read repo info via the API with an old URL. But all the info you get back refers to the current org. Which then leads usethis to think you have a weird remote setup. And there's no way to learn about repo moving and redirection via the API AFAIK.

That being said, I don't think this is a show stopper. People just need to update their URLs.

Reply:
Noted in text.